src/routes/auth.go

```diff
@@ -0,0 +1,39 @@
+package routes
+
+import (
+	"github.com/dana-team/platform-backend/src/types"
+	"github.com/dana-team/platform-backend/src/utils"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+// Login handles user authentication and issues a token on successful login.
+// It expects a JSON payload containing the user's credentials:
+//
+//	{
+//	   "username": "your_username", "password": "your_password"
+//	}
+//
+// It responds with a JSON object containing the token on successful login:
+//
+//	{
+//	  "token": "your_generated_token"
+//	}
+//
+// If there's an error during authentication, it responds with an appropriate error message.
+func Login() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var user types.User
+		if err := c.ShouldBindJSON(&user); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request payload"})
```

Comment:
Please use consider using `c.AbortWithStatusJSON(...)` because we'd like to abort it.